NDTensors/src/lib/Sectors/src/category_definitions/o2.jl

```diff
@@ -0,0 +1,64 @@
+#
+# Orthogonal group O(2)
+# isomorphic to Z_2 ⋉ U(1)
+# corresponds to to SU(2) subgroup with Sz conservation + Sz-reversal
+#
+
+struct O2 <: AbstractCategory
+  l::HalfIntegers.Half{Int}
+end
+
+SymmetryStyle(::O2) = NonAbelianGroup()
+
+category_label(s::O2) = s.l
+
+trivial(::Type{O2}) = O2(0)
+zero_odd(::Type{O2}) = O2(-1)
+
+_iszero(s::O2) = _iszero(category_label(s))
```

Comment:
Maybe call this `is_label_zero`?

Reply:
Or `istrivial`?

Reply:
Oh I see, it has another definition below that isn't `istrivial`. Though I would still like a slightly more descriptive name for this.

Reply:
`is_zero` matches both 0even and 0odd irreps. I added comments in the code to explain this. I can rename it to `is_zero_even_or_odd` for instance, but is this really needed? These are helper functions that are not designed to be public.

Reply:
A longer, more descriptive name would be better, `is_zero_even_or_odd` sounds good. I prefer to design code in a way where you keep it up to the standard of a public interface, even if it is originally expected to only be for internal usage, since then it is easier to maintain, read, and understand. Library code is more often read rather than edited, so designing it for making it easier to read and understand is best. For example, if this function had a more descriptive name to begin with, I maybe could have read the code and understood what it was doing, without having to ask you or look at the definition (you can tell I was confused by the name).